src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/PayRatesService.kt

```diff
@@ -97,6 +113,28 @@ class PayRatesService(
     return newPayRates
   }
 
+  /**
+   * See https://dsdmoj.atlassian.net/browse/SDIT-1605
+   */
+  private fun handleEndDate(
+    previousActivityEndDate: LocalDate?,
+    existingActivity: CourseActivity,
+    existingCurrentPayRate: CourseActivityPayRate?,
+  ) {
+    if (existingCurrentPayRate != null) {
```

Comment:
would be nicer with some sort of `existingCurrentPayRate?.` condition as four levels of `if` are difficult to read

Reply:
having merged the nested ifs (which i didnt know intellij can do automatically !) , im not sure existingCurrentPayRate?.let {...} is any clearer 🤷